airflow/executors/kubernetes_executor.py

```diff
@@ -265,12 +265,12 @@ def run(self):
 
     def _run(self, kube_client, resource_version, worker_uuid, kube_config):
         self.log.info(
-            'Event: and now my watch begins starting at resource_version: %s',
-            resource_version
-        )
+            'Event: and now my watch begins starting at resource_version: %s, '
+            'worker_uuid: %s', resource_version, worker_uuid)
         watcher = watch.Watch()
 
-        kwargs = {'label_selector': 'airflow-worker={}'.format(worker_uuid)}
+        kwargs = {'label_selector': 'airflow-worker={}'.format(worker_uuid),
+                  'timeout_seconds': 50}
```

Comment:
Yeah, this should be a config variable at the least.

Also it would be good if https://github.com/apache/airflow/blob/df35957fd4bab86ca6d2c923899569fb23139f56/airflow/config_templates/default_airflow.cfg#L782-L787 applied here (if it doesn't already?) so that this can be set globally for all Kube API calls form airflow.

Reply:
Wait, we already use the config option two lines down.

Someone on slack mentioned that this won't work but I don't see why from the code.

Reply:
If `kube_client_request_args` is used
```
[2019-11-25 18:02:53,397] {scheduler_job.py:1352} ERROR - Exception when executing execute_helper
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1350, in _execute
    self._execute_helper()
  File "/usr/local/lib/python3.7/site-packages/airflow/jobs/scheduler_job.py", line 1439, in _execute_helper
    self.executor.heartbeat()
  File "/usr/local/lib/python3.7/site-packages/airflow/executors/base_executor.py", line 136, in heartbeat
    self.sync()
  File "/usr/local/lib/python3.7/site-packages/airflow/contrib/executors/kubernetes_executor.py", line 801, in sync
    self.kube_scheduler.run_next(task)
  File "/usr/local/lib/python3.7/site-packages/airflow/contrib/executors/kubernetes_executor.py", line 456, in run_next
    self.launcher.run_pod_async(pod, **self.kube_config.kube_client_request_args)
  File "/usr/local/lib/python3.7/site-packages/airflow/contrib/kubernetes/pod_launcher.py", line 62, in run_pod_async
    resp = self._client.create_namespaced_pod(body=req, namespace=pod.namespace, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/kubernetes/client/apis/core_v1_api.py", line 6115, in create_namespaced_pod
    (data) = self.create_namespaced_pod_with_http_info(namespace, body, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/kubernetes/client/apis/core_v1_api.py", line 6148, in create_namespaced_pod_with_http_info
    " to method create_namespaced_pod" % key
TypeError: Got an unexpected keyword argument 'timeout_seconds' to method create_namespaced_pod
```

Reply:
I believe the correct argument name here is [`_request_timeout`]. I can't link the generated python API file as it is too large for github, but it's on line 6141 of https://github.com/kubernetes-client/python/blob/master/kubernetes/client/api/core_v1_api.py.

[This doc link for the is kube_client_request_args dead](https://github.com/apache/airflow/blob/df35957fd4bab86ca6d2c923899569fb23139f56/airflow/config_templates/default_airflow.cfg#L786). It also states;

> List of supported params in **kwargs are similar for all core_v1_apis, hence a single config variable for all apis

I feel like this is the wrong approach, as these setting should be configurable on a per request basis, but that's another matter and much more complex. For example [this label_selector argument](https://github.com/apache/airflow/blob/e51e1c770dad235b3fd8fdc330e44b83df8dcc4a/airflow/executors/kubernetes_executor.py#L273) would fail if passed to the `create_namespaced_pod` function.